### PR TITLE
E2e cap deploy

### PR DESCRIPTION
**Pull request by Rasadus03**
AKS and EKS updates

### CODE REVIEW

File: aks/README.md [hunk 1]
```diff
@@ -6,19 +6,21 @@
  export ARM_TENANT_ID=<Tenant Id>
  export ARM_SUBSCRIPTION_ID=<Subscription id>
  ```
- These env vars are used by the Terraform AzureRM provider to login to Azure with the configured SP credentials. Note the SP must have sufficient privileges to deploy resources in the subscription and must be created with `az ad sp create-for-rbac`, it cannot be created via the portal.
+ These env vars are used by the Terraform AzureRM provider to login to Azure with the configured SP credentials. 
+ Note the SP must have sufficient privileges, the created account must have at least contributor role on the tenant you may used az cli by running  `az ad sp create-for-rbac`or via the portal by creating app
```
Comment (satadruroy) [inline aks/README.md]: You might want to make some minor fixes on the run-on sentences above.

File: aks/aks-post-processing.sh [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-   az vm run-command invoke -g $MCRGNAME -n $i --command-id RunShellScript --scripts \
+   az vm run-command invoke -g $MCRGNAME -n $i --subscription $ARM_SUBSCRIPTION_ID --command-id RunShellScript --scripts \
    "sudo sed -i -r 's|^(GRUB_CMDLINE_LINUX_DEFAULT=)\"(.*.)\"|\1\"\2 swapaccount=1\"|' \
    /etc/default/grub.d/50-cloudimg-settings.cfg && sudo update-grub"
-   az vm restart -g $MCRGNAME -n $i
+   az vm restart -g $MCRGNAME -n $i --subscription $ARM_SUBSCRIPTION_ID
 done
-az logout
\ No newline at end of file
+ az logout
```
Comment (satadruroy) [inline aks/aks-post-processing.sh]: Actually, swap accounting is no longer required from 1.5.1 onwards (as long as the nodes themselves have swap disabled which is true on AKS now).

So, I'd suggest just removing post-processing.sh and removing the dependency on it from the `k8sresources.tf`.

File: aks/external-dns.tf [hunk 3]
```diff
@@ -17,26 +25,25 @@ resource "helm_release" "external-dns" {
...
-    
     set {
         name = "logLevel"
         value = "debug"
     }
 
+    set {
+        name = "azure.resourceGroup"
+        value = "${data.external.dns.result["resourceGroup"]}"
+    }
```
Comment (satadruroy) [inline aks/external-dns.tf]: Nice! I think having the resource group specified here again is an external-dns bug but no idea when they will fix it, if at all...

File: aks/le-prod-cert-issuer.yaml.template [hunk 4]
```diff
@@ -12,7 +12,7 @@ spec:
       # The secret that holds the generated private key used to communicate with Let's Encrypt
       name: letsencrypt-prod-account-key
     dns01:
-    providers:
+      providers:
```
Comment (satadruroy) [inline aks/le-prod-cert-issuer.yaml.template]: Ah, that infamous indentation issue which took us half a day to figure out... :-)

File: aks/variables.tf [hunk 5]
```diff
@@ -10,7 +10,7 @@ variable "node_count" {
     default = "1"
 }
 variable "machine_type" {
-    default = "Standard_DS4_v2"
+    default = "Standard_DS3_v2"
```
Comment (satadruroy) [inline aks/variables.tf]: Our recommendation is to use DS4_v2 though...on DS3 we have seen bunch of test failures on HA

File: eks/modules/services/k8s.tf [hunk 6]
```diff
@@ -15,4 +15,27 @@ provider "kubernetes" {
...
+    name = "scopedpersistent"
+  }
+  storage_provisioner = "kubernetes.io/aws-ebs"
+  reclaim_policy      = "Retain"
+  allow_volume_expansion = true
+  parameters = {
+    type = "gp2"
+    zone = "${var.storage_zones}"
+  }
+}
```
Comment (satadruroy) [inline eks/modules/services/k8s.tf]: Thanks! A non-default storage class config was missing...

File: aks/README.md (hunk 1 above)
Comment (Rasadus03) [reply aks/README.md]: I am fixing it :)

File: aks/aks-post-processing.sh (hunk 2 above)
Comment (Rasadus03) [reply aks/aks-post-processing.sh]: I will do that in my next PR with GKE :) after testing it on both GKE and AKS :)

File: aks/le-prod-cert-issuer.yaml.template (hunk 4 above)
Comment (Rasadus03) [reply aks/le-prod-cert-issuer.yaml.template]: Yes :)

File: eks/modules/services/k8s.tf (hunk 6 above)
Comment (Rasadus03) [reply eks/modules/services/k8s.tf]: You are always welcome :)

**Comment by satadruroy** [comment]
Hi Rania,
Looking at this again. The AKS/AWS hybrid stuff is in a separate directory with its own complete set of .tf files and won't interfere with the main e2e stuff, right? In that case, we're good to merge. I also pushed a branch for testing with local charts you might want to overlay those changes in here so people can play with nightlies.

**Comment by Rasadus03** [comment]
Hi Satadruroy,

It won't interfere at all with the other folders this one is used to setup
certmanager and external dns to use the AWS Route53 as a domain service, I
have another one as well for GKE using AWS :)

for the branch sure I will push my stuff there as well and BTW I have
removed in my repo the enablement of the swap account on AKS and tested it
as well as updates and fixes I did for GKE (that one I have not yet pushed
the removal of the swap account as I am still testing fixes for Stratos on
GKE :)).

thanks,
-rania

**Comment by satadruroy** [comment]
Rania,
For the local chart work I meant pulling in the changes here (not pushing these changes into that branch as that's just a temporary feature branch) but we can do that through a separate PR as well.

Let me know when you are finished with your changes here then I can approve/merge.

Thanks!

**Pull request merged**